Data/AVL/Tree.idr

```diff
@@ -213,6 +213,50 @@ findKey pred = foldr (\k,v,p => if pred v then Just k else p) Nothing
 findKeyOf : (Eq v) => v -> AVLTree h k v -> Maybe k
 findKeyOf value = findKey (== value)
 
+-- --------------------------------------------------------------------- [ Key ]
+
+||| A proof that some key is found in a Tree
+public export
+data Key : k -> Tree k v -> Type where
+  Here : Key key (Node key _ _ _)
+  InRight : (later : Key key r) -> Key key (Node _ _ _ r)
+  InLeft : (later : Key key l) -> Key key (Node _ _ l _)
+
+||| A wrapper to be used with AVLTree
+public export
+AVLKey : k -> AVLTree h k v -> Type
+AVLKey key avl = Key key (getWitness avl)
+
+||| An empty tree has no key
+noEmptyKey : {key : k} -> Key key Empty -> Void
+noEmptyKey Here impossible
+noEmptyKey (InRight _) impossible
+noEmptyKey (InLeft _) impossible
+
+Uninhabited (Key key Empty) where
+  uninhabited = noEmptyKey
+
+||| An item that is not in the root, not in the left child and not in the
+||| right child is not in the Tree at all
+nowhere : {key : k} -> {val : v} ->
+          {key' : k} -> {l : Tree k v} -> {r : Tree k v} ->
+          Not (key = key') -> Not (Key key l) -> Not (Key key r) ->
+          Not (Key key (Node key' val l r))
+nowhere notHere notInLeft notInRight Here = notHere Refl
```

Comment:
Is it possible to change the name of `nowhere` to something more meaningful such as `noKeyFound` or something similar? I am concerned that the implementation looks confusing with the lower case `nowhere`.